src/main/java/pl/wojciechkostecki/revel/service/MenuItemService.java

```diff
@@ -0,0 +1,57 @@
+package pl.wojciechkostecki.revel.service;
+
+import org.springframework.stereotype.Service;
+import pl.wojciechkostecki.revel.mapper.MenuItemMapper;
+import pl.wojciechkostecki.revel.model.Menu;
+import pl.wojciechkostecki.revel.model.MenuItem;
+import pl.wojciechkostecki.revel.model.dto.MenuItemDTO;
+import pl.wojciechkostecki.revel.repository.MenuItemRepository;
+
+import javax.persistence.EntityNotFoundException;
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Service
+@Transactional
+public class MenuItemService {
+    private final MenuItemRepository menuItemRepository;
+    private final MenuService menuService;
+    private final MenuItemMapper itemMapper;
+
+    public MenuItemService(MenuItemRepository menuItemRepository, MenuService menuService, MenuItemMapper itemMapper) {
+        this.menuItemRepository = menuItemRepository;
+        this.menuService = menuService;
+        this.itemMapper = itemMapper;
+    }
+
+    public MenuItem save(MenuItemDTO menuItemDTO){
+        MenuItem menuItem = itemMapper.toEntity(menuItemDTO);
+        Menu menu = menuService.findById(menuItemDTO.getMenuId())
```

Comment:
`Cannot find Menu with id: x` should be the message. Always include ids or some information about failed entity lookup or in any exception so its easy to debug in the future. Look at the entity and figure out why it broke.